asserts/snapasserts/validation_sets.go

```diff
@@ -75,6 +75,9 @@ type ValidationSetsValidationError struct {
 	// Sets maps validation set keys referenced by above maps to actual
 	// validation sets.
 	Sets map[string]*asserts.ValidationSet
+	// ExtraSets holds the sets added to the assertions set as extras. It's a subset
+	// of Sets.
+	ExtraSets []*asserts.ValidationSet
```

Comment:
I understand this is convenient but in isolation it looks strange. Either we keep a first class notion of old and new sets: by either having AddNew and AddOld, or Add and AddNew on ValidationSets. Or we find a way to pass this information around that doesn't involve the error touching the code here. Anyway even if we ultimately keep it, this probably should be a list of set keys, or Sets And NewSets should be distjoint.

Reply:
thinking a bit more, in case New makes sense in this context but Old is probably not great as it would sound that New replaces Old while they are in addition instead.